mailing_list/models.py

```diff
@@ -25,7 +25,14 @@ def _get_mailing_lists_by_section_id(self, canvas_course_id):
 
     def get_or_create_or_delete_mailing_list_by_address(self, address):
         try:
-            (_, canvas_course_id, section_id) = address.split('@')[0].split('-')
+            # if there is no section id in the address, the email is the class list email
+            # which has a different address format
+            if address.count('-') > 1:
+                (_, canvas_course_id, section_id) = address.split('@')[0].split('-')
+            else:
+                section_id = 'None'
+                (_, canvas_course_id) = address.split('@')[0].split('-')
+            logger.debug('course_id: %s, section_id: %s' % (canvas_course_id, section_id))
         except (AttributeError, ValueError):
             logger.error("Failed to parse address in get_or_create_or_delete_mailing_list_by_address %s", address)
             raise MailingList.DoesNotExist
```

Comment:
Can we switch this block over to using regular expressions?

```
try:
    m = re.search(settings.LISTSERV_SECTION_ADDRESS_RE, address)
    if m:
        canvas_course_id = m.group('canvas_course_id')
        section_id = m.group('section_id')
    else:
        m = re.search(settings.LISTSERV_COURSE_ADDRESS_RE, address)
        canvas_course_id = m.group('canvas_course_id')
        section_id = m.group('section_id')
except (IndexError, AttributeError):
    raise MailingList.DoesNotExist
```
And in lti_emailer/settings.base.py:
```
LISTSERV_SECTION_ADDRESS_RE = "^canvas-(?P<canvas_course_id>\d+)-(?P<section>\d+)@%s$" % LISTSERV_DOMAIN
LISTSERV_COURSE_ADDRESS_RE = "^canvas-(?P<canvas_course_id>\d+)@%s$" % LISTSERV_DOMAIN
```